test/listen.test.js

```diff
@@ -45,6 +45,15 @@ test('listen accepts a port, address, and callback', t => {
   })
 })
 
+test('listen accepts options and a callback', t => {
+  t.plan(1)
+  const fastify = Fastify()
+  t.tearDown(fastify.close.bind(fastify))
+  fastify.listen({ port: 0, host: 'localhost' }, (err) => {
```

Comment:
Can you test all options that we can pass here?

Reply:
Sure, I'll work on it

Reply:
@mcollina Options added, is it as expectation?